Core/Services/StitchCoreRemoteMongoDBService/Sources/StitchCoreRemoteMongoDBService/Sync/DataSynchronizer.swift

```diff
@@ -507,8 +551,57 @@ public class DataSynchronizer: NetworkStateDelegate, FatalErrorListener {
      - returns: the result of the remove many operation
      */
     func deleteMany(filter: Document,
-                    in namespace: MongoNamespace) -> DeleteResult? {
-        fatalError("\(#function) not implemented")
+                    options: DeleteOptions?,
+                    in namespace: MongoNamespace) throws -> DeleteResult? {
+        guard var nsConfig: NamespaceSynchronization = self.syncConfig[namespace] else {
+            throw StitchError.clientError(
+                withClientErrorCode: StitchClientErrorCode.couldNotLoadSyncInfo)
+        }
+        let lock = nsConfig.nsLock
+        lock.writeLock()
+        defer { lock.unlock() }
+
+        var eventsToEmit = [(BSONValue, ChangeEvent<Document>)]()
+        let localColl = try localCollection(for: namespace, withType: Document.self)
+
+        let idsToDelete = try localColl.find(filter).map { doc -> BSONValue? in
```

Comment:
[nit] Use `compactMap` here. This will save us the step of having to do the `guard` later on.